src/shaders/depth.frag

```diff
@@ -6,20 +6,35 @@ in highp vec2 textureCoordinates;
 #else
 in highp float depth;
 #endif
-
+#ifdef DEPTH_VISUALIZATER
+uniform highp float depthScaling;
+out highp vec4 fragmentColor;
+#else
 out highp float originalDepth;
+#endif
 
 void main() {
   #ifdef UNPROJECT_EXISTING_DEPTH
-  highp float depth = texture(depthTexture, textureCoordinates).r;
-  originalDepth =
-    #ifndef NO_FAR_PLANE_PATCHING
-    /* We can afford using == for comparison as 1.0f has an exact
-       representation and the depth is cleared to exactly this value. */
-    depth == 1.0 ? 0.0 :
+    highp float depth = texture(depthTexture, textureCoordinates).r;
+    #ifdef DEPTH_VISUALIZATER
+    highp float
     #endif
-    depthUnprojection[1] / (depth + depthUnprojection[0]);
+    originalDepth =
+      #ifndef NO_FAR_PLANE_PATCHING
+      /* We can afford using == for comparison as 1.0f has an exact
+         representation and the depth is cleared to exactly this value. */
+      depth == 1.0 ? 0.0 :
+      #endif
+      depthUnprojection[1] / (depth + depthUnprojection[0]);
   #else
-  originalDepth = depth;
+    #ifdef DEPTH_VISUALIZATER
+    highp float
+    #endif
```

Comment:
I feel these `#ifdef`s are a bit excessive. Declare the variable once above for both cases so you don't need to do this.

Thinking more about this, I feel like two *completely unrelated* shaders are mashed up together. They're not even exposed through the same C++ class. Since this is for an interactive visualization and not ML, one extra pass won't matter, so what about

1. unprojecting depth as usual using the original shader
2. then have a shader that's dedicated *just* to visualization, which then processes the unprojected depth

That way the `VisualizerShader` doesn't need the depth-related inputs and can be also easily adapted to visualizing any other rendered data, such as semantic IDs.

Reply:
>  a shader that's dedicated just to visualization

I used to have a dedicated shader. But it is different from what you suggested here -- dedicated to process the **unprojected** depth. My shader will still do both depth unprojection + processing like it does here. And then I thought you might say, well why not leverage the existing depth.frag to avoid code duplication?? Clearly my guess took the wrong direction this time. :)

Well, I guess the focus is that do I need the depth-related input?
Answer: it depends whether I need "visualization ONLY" or "visualization + interaction".
Your comments is based on the former, while I was thinking the latter.
However, I am actually considering adding depth in later (directly read from the `depthRenderTexture` and output to the `gl_FragDepth`) since it gives me the ability for "interaction".